docs/decisions/0049-agents-assistantsV2.md

```diff
@@ -0,0 +1,157 @@
+
+# Agent Framework - Assistant V2 Migration
+
+## Context and Problem Statement
+
+Open AI has release the _Assistants V2_ API.  This builds on top of the V1 _assistant_ concept, but also invalidates certain V1 features.  In addition, the _dotnet_ API that supports _Assistant V2_ features is entirely divergent on the `Azure.AI.OpenAI.Assistants` SDK that is currently in use.
+
+### Open Issues
+- **Streaming:** To be addressed as a discrete feature
+
+
+## Design
+
+Migrating to Assistant V2 API is a breaking change to the existing package due to:
+- Class definition changes
+- Underlying capability differences (e.g. `file-search` vs `retrieval`)
+- Underlying V2 SDK is version incompatible with V1 (`OpenAI` and `Azure.AI.OpenAI`)
+
+
+### Configuration Classes
+
+Specific configuration/setttings classes are introduced to support the ability to define assistant behavior at each of the supported articulation points (i.e. _assistant_, _thread_, & _run_).
+
+
+#### Assistant Definition
+
+The `OpenAIAssistantDefinition` was previously used only when enumerating a list of stored agents.  It has been evolved to also be used as input for creating and agent and exposed as a discrete property on the `OpenAIAssistantAgent` instance.
+
+This includes optional _execution_settings_ for defining default _run_ behavior.  Since these execution settings are not part of the remote assistant definition, they are persisted in the assistant metadata for when an existing agent is retrieved.
```

Comment:
Is this a subclass of the current `PromptExecutionSettings`? So one has the ability to control the `FunctionChoiceBehavior`?

Reply:
No, its not.  It corresponds to the core assistant model:  https://platform.openai.com/docs/api-reference/assistants/object

The behavioral settings are defined by the associated `When invoking an `OpenAIAssistantAgent` directly (no-chat), settings that only apply to a discrete run may be specified.  These settings are defined as `OpenAIAssistantInvocationSettings`: `OpenAIAssistantDefinition.ExecutionSettings`.

> Note: I'm working on dialing in the function-choice / tool-choice option.

Reply:
Looking a little deeper, I wonder if `ToolCallBehavior` isn't the class that might be extended here?